packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -243,6 +243,20 @@ abstract class WidgetsBindingObserver {
   void didChangeAccessibilityFeatures() {}
 }
 
+/// An abstract class for service extension registration.
+///
+/// As BindingBase is a singleton in Flutter framework, and runApp will always ensure there will be
+/// one WidgetsBinding instance. See [WidgetsFlutterBinding.ensureInitialized] for more.
+/// Instances inherited from ServiceExtensionBinding can add itself using [WidgetsBinding.addServiceExtensionBinding]
+/// and will be registered after [WidgetsBinding.initServiceExtensions].
+abstract class ServiceExtensionBinding {
+  /// Same as the name for [BindingBase.registerServiceExtension].
```

Comment:
Each class member's API docs exist independently on docs.flutter.io and must explain its own context. If someone gets linked to this field's API page, it should explain a bit what this whole thing is for.